### PR TITLE
docs(input): document numeric input md-maxlength validation

### LINKED ISSUE #5321: mdMaxlength: Always invalid for input type="number"
An input flagged with type="number" and md-maxlength="#" will always have a validation error.

Can be seen in this example under 'Errors': https://material.angularjs.org/latest/demo/input

Material Design v 1.0.0-rc1

### PR DESCRIPTION
**Pull request by DevVersion**
As discussed with @EladBezalel, we should not parse the numeric model to a string but mention  it in the docs.

Fixes #5321

### CODE REVIEW

**Comment by EladBezalel** [comment]
LGTM

File: src/components/input/input.js [hunk 1]
```diff
@@ -154,7 +154,8 @@ function labelDirective() {
  * allows you to build complex forms for data entry.
  *
  * @param {number=} md-maxlength The maximum number of characters allowed in this input. If this is
- *   specified, a character counter will be shown underneath the input.<br/><br/>
+ *   specified, a character counter will be shown underneath the input.<br/>
+ *   The `md-maxlength` validation is restricted for the use of inputs, which accepts strings.<br/><br/>
```
Comment (topherfangio) [inline src/components/input/input.js]: I might shorten this to

> **Note:** Only valid for text/string inputs (not numeric).

Comment (DevVersion) [inline src/components/input/input.js]: I like it :+1:
